server/signal.go

```diff
@@ -11,12 +11,12 @@ import (
 
 // Signal Handling
 func (s *Server) handleSignals() {
-	if s.opts.NoSigs {
+	if s.getOpts().NoSigs {
 		return
 	}
 	c := make(chan os.Signal, 1)
 
-	signal.Notify(c, syscall.SIGINT, syscall.SIGUSR1)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGUSR1, syscall.SIGHUP)
```

Comment:
I would not add this in until we have programmatic tests via API for all options we care about etc. and good test coverage. We should not expose to end users too soon IMO.